dht_bootstrap.go

```diff
@@ -166,3 +166,12 @@ func (dht *IpfsDHT) runBootstrap(ctx context.Context, cfg BootstrapConfig) error
 		return err
 	})
 }
+
+func (dht *IpfsDHT) BootstrapRandom(ctx context.Context) error {
+	return dht.randomWalk(ctx)
+}
+
+func (dht *IpfsDHT) BootstrapSelf(ctx context.Context) error {
```

Comment:
These `Bootstrap*()` methods can be consolidated into a single `Bootstrap(ctx context.Context, targets... peer.ID)`. If targets has len 0, we assume one random peer. If invoked with multiple targets, we launch a goroutine for each. The self case can be handled easily by doing `dht.Bootstrap(ctx, host.ID())`.

Reply:
See the other comment. Yes, a bootstrap that exposes a custom target is nice. But arbitrary consumers don't care, and want the semantics promised by these functions without having to dig around and track API changes (it's easy to not know how to generate a random ID, obtain the current ID, or even generate a bad random ID). See the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter). So a PR allowing a custom target might be exposed in the future, if we saw need for it (we won't).

Reply:
Ok, thinking about it, I buy your argument. Consolidating in a single `Bootstrap(ctx context.Context, targets... peer.ID)` method _could_ be too much voodoo.